src/macports1.0/macports.tcl

```diff
@@ -6584,8 +6590,28 @@ proc macports::get_parallel_jobs {{mem_restrict yes}} {
     variable buildmakejobs; variable os_platform
     if {[string is integer -strict $buildmakejobs] && $buildmakejobs > 0} {
         set jobs $buildmakejobs
-    } elseif {$os_platform eq "darwin" && $buildmakejobs == 0
-              && ![catch {sysctl hw.activecpu} cpus]} {
+    } else {
+        set jobs [macports::calc_auto_max_jobs $mem_restrict]
+    }
+    return $jobs
+}
+
+# Get actual number of parallel jobs based on indexjobs, which may
+# be 0 for automatic selection.
+proc macports::get_index_jobs {{mem_restrict yes}} {
```

Comment:
Perhaps `mem_restrict` should default to `no` here, eliminating need for `portindex` to do so.

Reply:
> Perhaps `mem_restrict` should default to `no` here, eliminating need for `portindex` to do so.

Fixed